src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h

```diff
@@ -1709,6 +1720,34 @@ struct YarpSensorBridge::Impl
             }
         }
 
+        if (controlBoardRemapperInterfaces.motors)
+        {
+            ok = controlBoardRemapperInterfaces.motors->getTemperatures(
+                controlBoardRemapperMeasures.motorTemperatureUnordered.data());
+
```

Comment:
Back in time @MSECode and @valegagge that this method returns false even if there is one sensor returning an error.  @MSECode and @valegagge is this the case? If yes it may be useful to use the `controlBoardRemapperInterfaces.motors->getTemperature(...)` method

Reply:
exactly. You can use the `getTemperature()` (without "s") method.
Wait for @valegagge to confirm but I'm confident and sure it is like that

Reply:
Hi @GiulioRomualdi ,
you are right, but please consider that the temperature functions adopt the same pattern as the other API: an API terminating with `s` returns false when one or more single-joint functions return an error. 

Currently, to verify the behavior of the temperature sensor in the robot, we are returning any read value, so we never return false. In case of errors, we just return a specific negative value as described in the [documentation](https://icub-tech-iit.github.io/documentation/temperature_sensors/software/dataflow/#error-handling).

To be completely honest, maybe, in the future,  this error handling will be improved in order to signal the error in a more user-friendly way: i.e. in case of error, we will return false
and will fill the diagnostic message with the correct description of the occurred problem.

In the end, let me suggest using the `getTemperature`for single joint and thinking about how to manage the case in which it returns false.: use the last value? use `n/a`? etc...

Reply:
We need to understand how to handle this situation. Currently, our framework requests feedback from all sensors every 100ms using the YARP interface (encoders, FTs, IMUs etc etc). As far as I know, there is no way to distinguish whether a call is made via RPC or via a statext.